### DISCUSSION TITLE
Improving the geometry interface

### DISCUSSION THREAD

**Discussion by Dumpstar56** [directus/directus#19079] [Feature Requests]
### Summary

When creating geometry data, I would like the interface to be more versatile and faster in use. 
1. With the map interface, the user could search for addresses or coordinates to navigate the map interface faster. Then they can still select the point, for example, by clicking on the map, but got there faster.
2. A new interface that lets the user create points by entering coordinates would be a good addition or alternative to improving the map interface.

### Basic Example

For option 1, I imagine something akin to the search function on openstreetmap.
For option 2, this could just be several data fields required for the geometry.

### Motivation

Creating geometry data takes a long time. Handling the map interface is slow if the user wants to select points far away from existing points. Visually searching the map while navigating and figuring out where to zoom requires geographical knowledge beyond the place you want to create geometry data for.

### Detailed Design

I apologize that I am not well-versed to answer this question.
Option 1: Create a button on the left of the map interface with a magnifying glass that expands into a text search field. Show results as openstreetmap does in a panel on the left to select the most appropriate find. Then zoom to that address.
Option 2: For a point geometry, have a new, non-map interface that is simply made up of two, three, or four float fields (depending in whether you would like to allow X Y, X Y Z, or X Y Z M coordinates).

### Must Have:
- An address or place search to navigate the map interface.
- A new interface that allows entering coordinate data directly.

### Should Have:
- a results panel to select between search results.

### Could Have:
- customization for the format (decimal vs minutes) and number (X Y vx X Y Z) of coordinate fields required

### Won't Have:
- No idea.

### Drawbacks

The implementation of an address search would require a connection to gazetteer data, e.g. of open street map, and that may be complex to implement.

### Alternatives

I am not sure how to answer this. Not improving the navigation of the map interface, in my mind at least, deters users from using it in the first place to create geometry data using Directus, rather than using external websites that provide the coordinates which are then entered into the backend DB by query.

### Adoption Strategy

Sorry, I do not have enough expertise to answer this.

**Comment by rijkvanzanten** [comment]
> With the map interface, the user could search for addresses or coordinates to navigate the map interface faster. Then they can still select the point, for example, by clicking on the map, but got there faster.

Entering a mapbox API key in Settings should unlock this already 🤔 Geo-lookup APIs aren't free unfortunately, so this third party API key is a requirement.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by musfandi** [comment]
Please add map panel in insight
